src/Storefront/Framework/SystemCheck/Util/SalesChannelDomainUtil.php

```diff
@@ -96,7 +99,16 @@ public function handleRequest(Request $request): StorefrontHealthCheckResult
 
         while ($redirectCount <= self::MAX_REDIRECTS) {
             $requestStart = microtime(true);
-            $response = $this->kernel->handle($currentRequest);
+            try {
+                // don't let the kernel catch errors, so we can handle them ourselves
+                $response = $this->kernel->handle($currentRequest, HttpKernelInterface::MAIN_REQUEST, false);
```

Comment:
```suggestion
                $response = $this->kernel->handle($currentRequest, catch: false);
```

especially on boolean parameter the named argument feature is quite useful